pkg/skaffold/instrumentation/export.go

```diff
@@ -283,3 +299,160 @@ func errorMetrics(ctx context.Context, meter skaffoldMeter, m metric.Meter, labe
 		unknownCounter.Record(ctx, 1, labels...)
 	}
 }
+
+type TraceExporterConfig struct {
+	writer    io.Writer
+	keyFS     http.FileSystem
+	keyPath   string
+	jaegerURL string
+}
+
+type TraceExporterOption func(te *TraceExporterConfig)
+
+func WithWriter(w io.Writer) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.writer = w
+	}
+}
+
+func WithKeyFS(keyFS http.FileSystem) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.keyFS = keyFS
+	}
+}
+
+func WithJaegerURL(jaegerURL string) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.jaegerURL = jaegerURL
+	}
+}
+
+func WithKeyPath(keyPath string) TraceExporterOption {
+	return func(teconf *TraceExporterConfig) {
+		teconf.keyPath = keyPath
+	}
+}
+
+func initTraceExporter(opts ...TraceExporterOption) (trace.TracerProvider, func(context.Context) error, error) {
+	teconf := TraceExporterConfig{
+		writer:    os.Stdout,
+		keyFS:     nil,
+		keyPath:   "/secret/keys.json",
+		jaegerURL: "http://localhost:14268/api/traces",
+	}
+
+	for _, opt := range opts {
+		opt(&teconf)
+	}
+
+	if _, ok := os.LookupEnv("SKAFFOLD_EXPORT_TO_STDOUT"); ok {
+		logrus.Debugf("using stdout trace exporter")
+		return initIOWriterTracer(teconf.writer)
+	}
+
+	switch os.Getenv("SKAFFOLD_TRACE") {
```

Comment:
Can we use the `OTEL_TRACES_EXPORTER` environment variable? https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#exporter-selection

Reply:
I am fine with changing our implementation to use OTEL_TRACES_EXPORTER vs SKAFFOLD_TRACE but I have some concerns:
1) It isn't clear to me if for example I use this env var w/ jaeger if that will interfere w/ the jaeger setup I have created (the url endpoint, etc.)
2) I think it is easier to rationalize when all skaffold env vars start w/ SKAFFOLD_* prefix, it might be confusing that skaffold uses this other env var which might be in the shell for other purposes/tools

Lmk if you prefer the OTEL_TRACES_EXPORTER env var and I can make the change.

Reply:
It looked like we'd get the same effect with `OTEL_TRACES_EXPORTER=jaeger` and `OTEL_EXPORTER_JAEGER_ENDPOINT=http://localhost:14268/`?

Reply:
These defaults are not working for me with the same setup I was using prior, I am not able to see any metrics in the jaeger UI using what I would think is the same setup via these env vars
```
OTEL_TRACES_EXPORTER=jaeger
OTEL_EXPORTER_JAEGER_ENDPOINT=http://localhost:14268/ # also tried http://localhost:14268/api/traces
```
I believe this will not work for the below reasons:

For the trace setup w/ jaeger we do create:
- a custom service "skaffold-trace"
- setup an sdk sampler w/ AlwaysSample

and in general we do the following:
- close all spans before os.Exit call via TracerShutdown callbacks

As such I will keep the SKAFFOLD_TRACE env var and recommend that be used.  Lmk and I can make the `OTEL_TRACES_EXPORTER` also make traceEnabled=true but currently it does not seem to work as I would expect

Reply:
I have made the OTEL_TRACES_EXPORTER env var make `traceEnabled=true` but I am having trouble verifying if they work properly

Reply:
My reasoning behind trying to leverage the OTEL*  variables is that we can then fit into existing systems.  We have some known partners (and a few unknown, I suspect) who may wish to integrate our traces with their own tracing.  They may be using something else other than Jaeger.

That said, the Go OpenTelemetry SDK for Go doesn't support OTEL_TRACES_EXPORTER (https://github.com/open-telemetry/opentelemetry-go/issues/1698).

So lets punt on this for now.